crates/kubelet/src/kubelet.rs

```diff
@@ -49,6 +55,97 @@ pub enum Phase {
 pub struct Status {
     pub phase: Phase,
     pub message: Option<String>,
+    pub container_statuses: Vec<ContainerStatus>,
+}
+
+/// A struct that contains all necessary information for a waiting status, meant
+/// to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct WaitingStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+    /// A human readable string describing the why it is in a waiting status
+    pub message: String,
+}
+
+/// A struct that contains all necessary information for a running status, meant
+/// to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct RunningStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+}
+
+/// A struct that contains all necessary information for a terminated status,
+/// meant to be used with the [ContainerStatus] enum
+#[derive(Clone, Debug)]
+pub struct TerminatedStatus {
+    /// The timestamp of when this status was reported
+    pub timestamp: DateTime<Utc>,
+    /// A human readable string describing the why it is in a terminating status
+    pub message: String,
+    /// Should be set to true if the process exited with an error
+    pub failed: bool,
+}
+
+/// ContainerStatus is a simplified version of the Kubernetes container status
+/// for use in providers. It allows for simple creation of the current status of
+/// a "container" (a running wasm process) without worrying about a bunch of
+/// Options. Use the [ContainerStatus::to_kubernetes] method for converting it
+/// to a Kubernetes API status
+#[derive(Clone, Debug)]
+pub enum ContainerStatus {
+    Waiting(WaitingStatus),
+    Running(RunningStatus),
+    Terminated(TerminatedStatus),
+}
+
+impl ContainerStatus {
+    pub fn to_kubernetes(&self, pod_name: &str) -> KubeContainerStatus {
+        let state = match self {
+            Self::Waiting(w) => ContainerState {
```

Comment:
I think it would be more readable to initialize a mutable `ContainerState` with all `Default::default` and then in each branch set the appropriate item.